### PR TITLE
Merge simple_dist_transpiler.py into distribute_transpiler.py

### LINKED ISSUE #11053: Update simple_distribute_transpiler.py
Distribute transpiler will transpile the forward pass code by user to the distributed code version, what transpiler does was listed below:
```
Steps to transpile trainer:                                                                                                           
 1. split variable to multiple blocks, aligned by product(dim[1:]) (width)                                                                                             
 2. rename splited grad variables to add trainer_id suffix ".trainer_%d".                                                              
 3. modify trainer program add split_op to each grad variable.                                                                         
 4. append send_op to send splited variables to server and fetch params(splited blocks or origin param) from server.                                                                               
 5. append concat_op to merge splited blocks to update local weights.  
```

and the simple distribute transpiler did the similar things which was listed below
```
Steps to transpile trainer:                                                            
 1. rename splited grad variables to add trainer_id suffix ".trainer_%d".                                                               
 2. append send_op to send splited variables to server and fetch params
```

so, actually the simple version could be merged into the original files.

### PR DESCRIPTION
**Pull request by velconia**
This close #11053

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_simple_dist_transpiler.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import unittest
+
+import paddle.fluid as fluid
+import paddle.fluid.core as core
+import paddle.fluid.layers as layers
+from paddle.fluid.transpiler.distribute_transpiler import delete_ops
+import numpy as np
+
+
+class TestSimpleDistTranspiler(unittest.TestCase):
```
Comment (typhoonzero) [inline python/paddle/fluid/tests/unittests/test_simple_dist_transpiler.py]: This test can inherit from `TestDistTranspiler` and just make some small changes

Comment (velconia) [reply python/paddle/fluid/tests/unittests/test_simple_dist_transpiler.py]: Thanks for reviewing, I make a base unittest class in the next commit

File: python/paddle/fluid/transpiler/distribute_transpiler.py [hunk 2]
```diff
@@ -196,8 +197,15 @@ def _init_splited_vars(self, split_method):
         self._update_dist_lookup_table_vars(param_list, grad_list,
                                             self.params_grads)
 
-        grad_blocks = split_variable(grad_list, len(self.pserver_endpoints))
-        param_blocks = split_variable(param_list, len(self.pserver_endpoints))
+        if align_var_to_block:
```
Comment (typhoonzero) [inline python/paddle/fluid/transpiler/distribute_transpiler.py]: That's smart! `align_var_to_block` this argument may be changed to some name easier to understand? like `do_slice`?

Comment (typhoonzero) [reply python/paddle/fluid/transpiler/distribute_transpiler.py]: Sounds like `is_slice_var` is to say is this variable sliced?

Comment (velconia) [reply python/paddle/fluid/transpiler/distribute_transpiler.py]: How about ```do_var_slice_up```

Comment (velconia) [reply python/paddle/fluid/transpiler/distribute_transpiler.py]: I changed the variables name and the logic of _init_splited_vars, please review it

**Pull request merged**